### PR TITLE
Fix same deposits from same validator in same block

### PR DESCRIPTION
**Pull request by terencechain**
We process deposits in a block by precomputing `valIndexMap`, that is to check if the validator has already existed in the state, but we only precompute `valIndexMap` once before processing all the deposits in the body. This doesn't cover deposits from the same validator in the same block

Change list
- [x] Update `valIndexMap` on per deposit
- [x] Generate deposit and trie helper for regression test of the multiple deposits from same validator
- [x] Regression test

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Does this resolve #4218?

**Comment by terencechain** [comment]
> Does this resolve #4218?

No it doesn't. This resolves duplicated validator entires in state

File: beacon-chain/core/blocks/block_operations_test.go [hunk 1]
```diff
@@ -1343,6 +1343,46 @@ func TestValidateIndexedAttestation_AboveMaxLength(t *testing.T) {
...
+			PreviousVersion: params.BeaconConfig().GenesisForkVersion,
+			CurrentVersion:  params.BeaconConfig().GenesisForkVersion,
+		},
+	}
+	newState, err := blocks.ProcessDeposits(context.Background(), beaconState, block.Body)
+	if err != nil {
+		t.Fatalf("Expected block deposits to process correctly, received: %v", err)
+	}
+
+	if len(newState.Validators) > 2 {
```
Comment (mcdee) [inline beacon-chain/core/blocks/block_operations_test.go]: Why not `!= 2` here to ensure the validator is added?

Comment (rauljordan) [reply beacon-chain/core/blocks/block_operations_test.go]: +1

Comment (terencechain) [reply beacon-chain/core/blocks/block_operations_test.go]: Agree. Also test is failing. Looking into it now

**Comment by prestonvanloon** [comment]
This is consensus breaking and confirmed to have happened in this testnet already.

`0xabcee7d41a591b96a0183dc06eda2d5819f1028e531b51a511dc0098fe2f41eeb40ffb8702b769a0cc96227e6a9c6462` is present in the validator registry three times.

**Pull request merged**